### ISSUE TITLE
New Device Request - Sage Doorbell Sensor

### ISSUE THREAD

**Issue by Puntoboy** [dresden-elektronik/deconz-rest-plugin#2755]
Hello, I've recently bought this sensor and I would like to integrate it with my Home Assistant. Below are what I think are the required screenshots.

![image](https://user-images.githubusercontent.com/11718878/81484783-5e8bbd80-9240-11ea-8e78-d5206b4c81aa.png)

![image](https://user-images.githubusercontent.com/11718878/81484786-68adbc00-9240-11ea-9b17-89fb022dc418.png)

**Comment by MattL0** [comment]
+1

**Comment by MattL0** [comment]
Just ordered mine.

**Comment by Puntoboy** [comment]
I've only bought this sensor as my video doorbell won't send notification to Alexa so I want to use HA to do it instead.

**Comment by MattL0** [comment]
is there any new hardware integration on deconmz? it is calm since a while @manup

**Comment by MattL0** [comment]
Up @manup ?

**Comment by Mimiix** [comment]
@SwoopX

**Comment by ManicGTI** [comment]
+1 here as well, this would be useful for my Nest Hello

**Comment by Puntoboy** [comment]
Any update on this?

**Comment by MattL0** [comment]
+1 would really like to add this to my setup @manup

**Comment by SwoopX** [comment]
I have no clue if this will work since I don't have the device. Chances are very high that the button map is not correct. I'd require some debug output then.

**Comment by Puntoboy** [comment]
> 
> 
> I have no clue if this will work since I don't have the device. Chances are very high that the button map is not correct. I'd require some debug output then.

Tell me what you need and I can send it. You'll have to break it down though as I am new to this.

**Comment by SwoopX** [comment]
Dunno, they did it with 2 buttons, I do it with one. Should have pretty much the same effect

**Comment by Puntoboy** [comment]
So what I've found is that I can pair it to Deconz, it appears in the VNC viewer but not in the web app. When I look at the logs in HA there is no input from the device.

**Comment by SwoopX** [comment]
> When I look at the logs in HA there is no input from the device.

No surprise since it's not supported yet.

**Comment by Puntoboy** [comment]
> 
> 
> > When I look at the logs in HA there is no input from the device.
> 
> No surprise since it's not supported yet.

Clearly...


So where do we go from here?

**Comment by SwoopX** [comment]
Wait for the new release and check it out. Changed the button map over here to mimic zha. I feel the device could behave funny otherwise (and possibly brick the intended functionality). Worst case is that you'd require 2 buttons for your home automation, 1 for open and one for close.

**Comment by Puntoboy** [comment]
Great thanks. I'll give it a try.

**Comment by SwoopX** [comment]
Closing this as support has been added with version .79

**Comment by ManicGTI** [comment]
Whoop!  Thanks.

**Comment by Puntoboy** [comment]
Pity I can’t upgrade to get it :(

Kind regards,
Chris Jeffrey

Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Comment by SwoopX** [comment]
I need to have some feedback if it works alright though. If not, please raise a seperate issue for that :wink:

**Comment by ReX1983** [comment]
> I need to have some feedback if it works alright though. If not, please raise a seperate issue for that 😉

Good stuff @SwoopX: it works (integration in Home Assistant).
I have connected only the front wire (it has also a rear wire) and what I get in Home Assistant is event 1002 when the doorbell button is pressed. There is no release event, but that's fine.

For those interested: it works also with Nest Hello. I don't have an internal chime, was hoping to use the Nest notifications played on a Google speaker but it has too much delay to be used in a real life scenario.

**Comment by ManicGTI** [comment]
Excellent, It's the Nest Hello I want to use it with.  Have you hooked it up via the Chime Connector?  I'm assuming I can run it in parallel with a chime when I eventually get one.

**Comment by ReX1983** [comment]
> Excellent, It's the Nest Hello I want to use it with. Have you hooked it up via the Chime Connector? I'm assuming I can run it in parallel with a chime when I eventually get one.

I've used the Chime connector even if I don't have any chime. The wiring is: one white/grey couple from the chime connector is wired to the zigbee sensor and the other couple is in series with the power source and Nest Hello.

The chime connector ensures that the zigbee sensor is not constantly powered (that would be the case if installed in series or parallel of Nest Hello, without the connector).

My theory is that the chime connector detects the button press (by measuring the tension on the series) and powers the other line connected to the chime or the zigbee sensor.

**Comment by Puntoboy** [comment]
I would love to start testing this but the newer versions of Deconz won't work with my HA install on Docker. :( So until I move my HA/Deconz install to a different system I cannot update.

**Comment by Mimiix** [comment]
@Puntoboy If you are on discord, we can take a look together?

**Comment by Puntoboy** [comment]
> 
> 
> @Puntoboy If you are on discord, we can take a look together?

Thanks but I'm not on Discord. 

Once I move my HA install from Docker on Synology I'll be able to update the Deconz install. I had already upgraded but then found that Deconz stopped working. I posted here about it https://github.com/home-assistant/hassio-addons/issues/1503#issuecomment-665001517

**Comment by Mimiix** [comment]
IT is useful to be there ;)

Anyway, I have deconz on a VM with HA. My advice would be run a VM with HA Supervised (https://github.com/DubhAd/HA-Notes/blob/main/supervised-on-linux.md) . Then run a second VM with raspbian/ubuntu and run deCONZ there.

Way more stable in my case.

**Comment by Puntoboy** [comment]
> 
> 
> > Excellent, It's the Nest Hello I want to use it with. Have you hooked it up via the Chime Connector? I'm assuming I can run it in parallel with a chime when I eventually get one.
> 
> I've used the Chime connector even if I don't have any chime. The wiring is: one white/grey couple from the chime connector is wired to the zigbee sensor and the other couple is in series with the power source and Nest Hello.
> 
> The chime connector ensures that the zigbee sensor is not constantly powered (that would be the case if installed in series or parallel of Nest Hello, without the connector).
> 
> My theory is that the chime connector detects the button press (by measuring the tension on the series) and powers the other line connected to the chime or the zigbee sensor.

Thanks very much for this. I have finally migrated my HA installation to a virtual machine and upgraded Deconz to the latest version, I now have the Sage sensor working with Node Red and using Alexa to announce when the doorbell is ringing :) Handy when I'm away from the chime in the garage or something. :)

**Comment by ReX1983** [comment]
> Thanks very much for this. I have finally migrated my HA installation to a virtual machine and upgraded Deconz to the latest version, I now have the Sage sensor working with Node Red and using Alexa to announce when the doorbell is ringing :) Handy when I'm away from the chime in the garage or something. :)

Good stuff!

I have an automation to play a sound on the Google Home speakers and one automation to play a sound on the Xiaomi gateways. Eventually I have decided to stick with the latter because it all works in local: it is faster and more reliable.

**Comment by MattL0** [comment]
Thanks for this .

Is it possible to add the battery attribute? I only get null.

**Comment by MattL0** [comment]
I do not see it in phoscon, but I can acess it from the Homeseer or json api. 

 all i get is the 1002 value when pressed.. Is that ok? 

thanks

**Comment by SwoopX** [comment]
I assume not, but I leave it for the moment. If all goes well, you as users can make a change, if required, on your own without touching the code.

**Comment by Puntoboy** [comment]
Yes this is another device that doesn't show in the webapp but does show in HA. Mine is working fine. Just using the front connection so have it in node red to send an announcement via Alexa. 

![image](https://user-images.githubusercontent.com/11718878/90337225-01d73600-dfd9-11ea-93ff-2b9d6a88027b.jpeg)

**Comment by Chr1stian** [comment]
How do you use the sensor when connected trough DeCONZ? It shows as "Bell" for me in HA, but no entities and listening to `deconz_event` reports nothing when I put voltage on the pins of the sensor :/ Appreciate any help on the setup process of Sage + Conbee II + DeCONZ + HA 🙏

**Comment by MattL0** [comment]
Anyone knows how long the battery will last with this?

**Comment by MattL0** [comment]
We just moved to a new place.  I reinstalled the sage doorbell, but it was always ''unreachable'' and it didn't send anything to jowihue homeseer plugin. 

I tried to pair a brand new sage doorbell but got the same result. 

This had been flawless since the almost a year ( I added the sensors at that time and never touched it) . I do not understand . I also tried to remove any trace of this sensor in the zll.db  and repair it again, but no luck. Still not reachable and does not send anything to homeseer when the doorbell is pressed. 

anyone have any insight?

**Comment by MattL0** [comment]
**************************** Object info *****************************
{
"config": {
"group": "2",
"on": true,
"reachable": false
},
"ep": 18,
"etag": "89b725909280f2e73ba5e0c2d299c305",
"lastseen": null,
"manufacturername": "Echostar",
"mode": 1,
"modelid": "Bell",
"name": "Doorbell Sensor",
"state": {
"buttonevent": null,
"lastupdated": "none"
},
"swversion": "3.2.2",
"type": "ZHASwitch",
"uniqueid": "00:08:89:00:01:d9:dc:13-12-0006"
}

**Comment by MattL0** [comment]
i 'll open a new issue for this

**Comment by oywino** [comment]
> We just moved to a new place. 
> Anyone have any insight?
I've read that you need at least 12VDC to trigger the sensor, so if your doorbell at the place differs from the one you had at the old place, then that might be the reason.

**Comment by oywino** [comment]
> i 'll open a new issue for this

Did you ever manage to solve this issue?
I'm having exactly the same problem.  Pairing it using Phoscon was easy, and it shows up in Phoscon like this:
![image](https://user-images.githubusercontent.com/18447119/114318823-0542a300-9b0f-11eb-85c3-07d9aea93a2f.png)
But as you can see, it's reported as "unreachable".
In HA, it shows up as a "Device" with no "Entities" and there's nothing on the event bus.
![image](https://user-images.githubusercontent.com/18447119/114318892-4f2b8900-9b0f-11eb-95e6-f6beaec6966f.png)

I'm running Marthoc/deConz 2.10.4 in a Docker Container with a connected ConBee II with Firmware 0x026660700
I need all the help I can get on this one 😊

**Comment by Adjrobinson** [comment]
I bought one of these and can't get it to show up in the phoscon webpage, I think this is it when I vnc in?
![image](https://user-images.githubusercontent.com/1196592/115076550-a016f500-9ef4-11eb-9ab0-44ef3d93e799.png). I would be happy to buy one and send it to the dev's as I would love to have it working.

**Comment by oywino** [comment]
This is what mine looks like in deConz;
![image](https://user-images.githubusercontent.com/18447119/115085921-9f3d8d80-9f0b-11eb-85b5-4173acbb4a52.png)

Can you check what firmware version is in your DoorBell sensor?
![image](https://user-images.githubusercontent.com/18447119/115086202-170bb800-9f0c-11eb-8b34-0a5286eee0eb.png)

**Comment by Adjrobinson** [comment]
It doesn't seem to show: 
I tried deleting it and adding it again with no difference.
![image](https://user-images.githubusercontent.com/1196592/115091445-656e8600-9f0f-11eb-8bb1-3d742cd7ade2.png)

**Comment by oywino** [comment]
I think you have to click in the field "SW Build ID" to bring up the Attribute Editor, and in this click on "Read" (or Read Config) - I don't remember exactly.

**Comment by Adjrobinson** [comment]
ah yes, looks like the same, weird, I deleted it again and turned off the routers and paired it in the same room as the combee, still no joy: 
![image](https://user-images.githubusercontent.com/1196592/115104457-7e4b5b80-9f50-11eb-969b-1595eb09790a.png)

**Comment by Mimiix** [comment]
It's normal it doesn't show in phoscon. They need to add support for it.

**Comment by Adjrobinson** [comment]
Ok, I have ordered https://www.tindie.com/products/ErikLemcke/doorbell-modernizr-2/ instead so if one of the dev's want the sage one to add it let me know as it's pretty pointless at the moment lol, looks like zigbee2mqtt might support it but I don't want to go down that route.

**Comment by oywino** [comment]
> It's normal it doesn't show in phoscon. They need to add support for it.

Incorrect. Mine does show in Phoscon as well as in Home Assistance! It's just listed an "unreachable". Using deConz, I can read attribute values, but I cannot activate it. But I have read elsewhere that others have been able to make it work using deConz until they one day disconnected it and could no longer successfully reconnect it again after updating.

**Comment by Smanar** [comment]
Deconz version 2.05.77, idk in wich one version this device was added, but can you try with a more recent version ?

**Comment by oywino** [comment]
I'm using deConz 2.10.04.
I'd love to share my JSON but I don't know how. Please explain.

**Comment by Smanar** [comment]
You can find it.
- using third app
- using phoscon with dev console
- checking direclty the api

`http://IP:PORT/api/KEY/lights`
and
`http://IP:PORT/api/KEY/sensors`

IP and PORT are the same than for phoscon
Key is an api key https://dresden-elektronik.github.io/deconz-rest-doc/getting_started/

**Comment by oywino** [comment]
Here it is:

`"12": {
        "config": {
            "group": "20",
            "on": true,
            "reachable": true
        },
        "ep": 18,
        "etag": "75031fec22a9760f05a1604edebac11b",
        "lastseen": "2021-04-17T06:19Z",
        "manufacturername": "Echostar",
        "mode": 1,
        "modelid": "Bell",
        "name": "Doorbell Sensor",
        "state": {
            "buttonevent": null,
            "lastupdated": "none"
        },
        "swversion": "3.2.2",
        "type": "ZHASwitch",
        "uniqueid": "00:08:89:00:01:d9:af:d2-12-0006"
    },`

**Comment by Smanar** [comment]
Oups sorry, I have read "doorbell" so in my mind was a "siren" not a button.

Do you have access to the GUI ? I need log when you press the button, deconz/help/log with "info" and "info_l2"

**Comment by oywino** [comment]
Well, that's the whole problem. Because it appears as "unreachable", nothing happens when I press the button.
We first have to make it appear correctly and reachable in deConz.

**Comment by Smanar** [comment]
No I don't say, phoscon but deconz, the application.

**Comment by Puntoboy** [comment]
Sorry I haven’t read all the replies but my sage doorbell is showing in Phoscon and working great with HA. I have it triggering Alexa when someone rings the doorbell.

Kind regards,
Chris Jeffrey

Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Comment by oywino** [comment]
> Sorry I haven’t read all the replies but my sage doorbell is showing in Phoscon and working great with HA. 

Could you please share the Entity details that the Sage Doorbell is giving you in HA ?

**Comment by Puntoboy** [comment]
Not sure what’s that going to show you?

[Image.jpeg]

[Image.jpeg]


Kind regards,
Chris Jeffrey

Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Comment by oywino** [comment]
(You forgot the images)

**Comment by Puntoboy** [comment]
I didn't :) just the email didn't upload them. 

![image](https://user-images.githubusercontent.com/11718878/115155629-c6659d80-a078-11eb-8b4e-ff4052839adc.png)

![image](https://user-images.githubusercontent.com/11718878/115155632-ca91bb00-a078-11eb-9332-4d39bc39bb45.png)

**Comment by oywino** [comment]
Well, thank you, but to me, that looks like the "Device" entry in HA followed by the "sensor" entry from Phoscon, right?
What I was hoping to see, is the "Entity" entry from HA.  And please, don't use mobile phone images. I'd much rather see HA images from a full browser (PC or MAC)  🎵😊

**Comment by Puntoboy** [comment]
Well I’m on my phone so that’s all I have. The entity in HA is how that first screenshot shows.

Kind regards,
Chris Jeffrey

Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Comment by Puntoboy** [comment]
There is no entity. 

![image](https://user-images.githubusercontent.com/11718878/115160547-b1950400-a090-11eb-8111-bd7a34ddedf2.png)

**Comment by oywino** [comment]
Exactly, that's what I don't see either.  So you read the state of the bell into automations using deconz.event then?
Did you manage to decipher the event codes?

**Comment by Puntoboy** [comment]
Ah yes sorry, I use Node-Red in HA to run the automation. It uses a direct connection to Deconz.

![image](https://user-images.githubusercontent.com/11718878/115202220-17fc3f80-a0ee-11eb-8b7c-675631289d1d.png)

**Comment by oywino** [comment]
Ok, I see. But still, it is the deconz.event node that you are using.
 My problem is that the DoorBell sensor appear as "unreachable" in Phoscon, and consequently it never generates any events,
I am puzzled by the fact that the in JSON it is tagged as "reachable" while in Phoscon it is "unreachable".
I assume that's not the case with your device?

**Comment by Puntoboy** [comment]
The sensor appears unreachable in mine as well, but I'm using the switch device (see the first screenshot I shared yesterday).

**Comment by Puntoboy** [comment]
This is the switch device.

![image](https://user-images.githubusercontent.com/11718878/115206870-f6ea1d80-a0f2-11eb-877b-0f9dae70489c.png)

**Comment by oywino** [comment]
Oh, thank you 👍 That was very interesting.
I now re-paired the sensor after upgrading deconz to v2.11 and now I also have two devices; A sensor and a switch. This is slowly moving forward thanks to your help 😊
Do you know the specs for your bell transformer?

**Comment by Puntoboy** [comment]
The transformer was already installed in our house. It's this one https://www.amazon.co.uk/MDC1-01-Traditional-Chime-Built-Transformer/dp/B07CZMBKNH

**Comment by Smanar** [comment]
So the device appear in "sensors" AND "switches" but there is only 1 Json ?

**Comment by oywino** [comment]
Only one that I could find.
deConz JSON only supports `lights` and `sensors`
Switches are not supported. They appear as `sensors`

**Comment by Smanar** [comment]
Ha, ok, it s normal, switches is something from phoscon.
